src/main/java/com/rackspace/ceres/app/web/QueryController.java

```diff
@@ -99,7 +101,7 @@ public QueryController(QueryService queryService, DownsampleProperties downsampl
       @RequestParam List<String> tag,
       @RequestParam String start,
       @RequestParam(required = false) String end) {
-    validateMetricNameAndMetricGroup(metricName, metricGroup);
+    validator.validateMetricNameAndMetricGroup(metricName, metricGroup);
```

Comment:
Why not create a more generic validator that can take a list of strings? Then we don't need these very long method names...

Reply:
here, we need to validate metricName or metricGroup as both must not be empty and both must not be non-empty as well.

Reply:
You can creata generic validator for strings that shold not be empty/null etc. And to make it even more generic you can have it take a list of strings so you can validate 1/2/3/... number of strings at the same time.